### ISSUE TITLE
question regarding the optimizer

### ISSUE THREAD

**Issue by seyeeet** [facebookresearch/3detr#26]
Hi Ishan
Thanks a lot for such an amazing work.
I have a question and appreciate it if you can please provide your insight and opinion on that.
I see in this work and a couple of your other works you used adamw, I notice that the parameters for the adamw is quite different from the ones that we use for adam though.
- I was wondering why is that and if there is a reason behind it?

My other question is that
- I notice that there is no stablished parameters for adamw yet, in different works of you I notice different parameters for adamw. I would be really thankful if you can provide some suggestion on what should I take into account to choose the right hyper parameters for adamw and if you have any additional suggestion for that matter.
Thanks a lot for ur help :)

**Comment by imisra** [comment]
Thanks for your question @seyeeet 

Adam and AdamW differ in the way weight decay is handled. In AdamW, the weight decay is only proportional to the weight itself (and doesn't end up accounting for the moving averages). So, weight decay values will look significantly different across Adam and AdamW.

I can tell you how I searched for learning rates. I started with the highest initial learning rate (and weight decay = 0) which did not lead to a diverging loss. This also meant using a linear warmup was important, otherwise the loss would diverge very frequently. Once I had this learning rate, I would search in a finer grid around it (and set weight decay > 0).

Hope this helps!
